examples/impute/plot_multiple_imputation.py

```diff
@@ -0,0 +1,359 @@
+    ----------
+    X : ndarray, shape (n_samples, n_features)
+        The data without missing samples.
+    missing_rate : float, default=0.75
+        The amount of missing data in ``X``.
+    strategy : {'MCAR', 'NMCAR'}, default='MCAR'
+        The strategy for removing data. Must be 'MCAR' or 'NMCAR'.
+        If MCAR, each sample will have a random missing feature.
+        If NMCAR, each sample will have a random missing feature. However,
+        values that are farther from the feature mean are more likely to be
+        made missing.
+
+    Returns
+    -------
+    X_out : ndarray, shape (n_samples, n_features)
+        Output data with missing entries.
+    """
+    X_out = X.copy()
+    n_drop_per_feature = int(missing_rate * X.shape[0])
+    for x in np.transpose(X_out):
+        if strategy == 'MCAR':
+            prob = None
+        elif strategy == 'NMCAR':
+            weights = special.expit(np.abs(x - x.mean()))
+            prob = weights / weights.sum()
+        drop_idx = np.random.choice(
+            X.shape[0], p=prob, replace=False, size=n_drop_per_feature
+        )
+        x[drop_idx] = np.nan
+    return X_out
+
+
+def variance_model_coef(y_true, y_pred, X):
+    """Calculates variance of the coefficients."""
+    residual_sum_squares = np.sum((y_true - y_pred) ** 2)
+    residual_mean_square = residual_sum_squares / (len(y_true) - 2)
+    return residual_mean_square / (X ** 2).sum(axis=0)
+
+
+def rubins_pooling_rules(m_coefs, m_vars_coefs):
+    """Applies Rubin's pooling rules.
+
+    The final weights is defined as the mean of the weights across the imputed
+    datasets while the total variance is defined as the combination of the mean
+    of the variance of the weights and the variance of the coefficients.
+
+    Parameters
+    ----------
+    m_coefs : ndarray, shape (n_imputations, n_features)
+        The weights of the model fitted on each imputed dataset.
+    m_vars_coefs : ndarray, shape (n_imputations, n_features)
+        An estimate of the variance of the weights on each imputed dataset.
+
+    Returns
+    -------
+    mean_coefs : ndarray, shape (n_features,)
+        The mean coefficients computed across the imputed datasets.
+    total_var_coefs : ndarray, shape (n_features,)
+        An estimate of the total variation of the weights across the imputed
+        datasets.
+    """
+    mean_coefs = np.mean(m_coefs, axis=0)
+    mean_vars_coefs = np.mean(m_vars_coefs, axis=0)
+    vars_coefs = np.var(m_coefs, axis=0, ddof=1)
+    total_var_coefs = (mean_vars_coefs +
+                       (1 + 1 / vars_coefs.shape[0]) * vars_coefs)
+    return mean_coefs, total_var_coefs
+
+
+###############################################################################
+# EXAMPLE 1. COMPARE STATISTICAL ESTIMATES AND THEIR VARIANCE USING MULTIPLE
+# IMPUTATION IN A LINEAR REGRESSION MODEL.
+
+MAX_ITER = 20  # number of iterations in IterativeImputer
+ESTIMATOR = Pipeline(steps=[('scaler', StandardScaler()),
```

Comment:
all capital case variable names for a class instance appears less common for the sklearn ecosystem...

Reply:
I'm doing this because it's a global for the entire file, but I'm open to other styling.